test/src/main/java/org/apache/accumulo/test/functional/ZooCacheIT.java

```diff
@@ -81,4 +100,149 @@ public void test() throws Exception {
+            return;
+          }
+          // LOG.info("received event {}", event);
+
+          String tableId = tableIdSup.get().get();
+
+          switch (event.getType()) {
+            case ChildWatchRemoved:
+            case DataWatchRemoved:
+            case NodeChildrenChanged:
+            case NodeDataChanged:
+            case PersistentWatchRemoved:
+              break;
+            case NodeCreated:
+              if (eventPath.equals(zooRoot + Constants.ZTABLES + "/" + tableId)) {
+                LOG.info("Setting tableCreatedEvent");
+                tableCreatedEvent.set(true);
+              }
+              break;
+            case NodeDeleted:
+              if (eventPath.equals(zooRoot + Constants.ZTABLES + "/" + tableId)) {
+                LOG.info("Setting tableDeletedEvent");
+                tableDeletedEvent.set(true);
+              }
+              break;
+            case None:
+              switch (event.getState()) {
+                case AuthFailed:
+                case ConnectedReadOnly:
+                case NoSyncConnected:
+                case SaslAuthenticated:
+                case Unknown:
+                  break;
+                case Closed:
+                case Disconnected:
+                  LOG.info("Setting connectionClosedEvent");
+                  connectionClosedEvent.set(true);
+                  break;
+                case Expired:
+                  LOG.info("Setting sessionExpiredEvent");
+                  sessionExpiredEvent.set(true);
+                  break;
+                case SyncConnected:
+                  LOG.info("Setting connectionOpenEvent");
+                  connectionOpenEvent.set(true);
+                  break;
+                default:
+                  break;
+
+              }
+              break;
+            default:
+              break;
+
+          }
+        }
+
+      });
+
+      assertFalse(connectionOpenEvent.get()); // missed the event, connection already established
+
+      client.tableOperations().create(tableName);
+      final String tid = ctx.tableOperations().tableIdMap().get(tableName);
+      tableId.set(tid);
+      // we might miss the table created event, don't check for it
+      final String tableZPath = zooRoot + Constants.ZTABLES + "/" + tid;
+      assertFalse(cache.childrenCached(tableZPath));
+      assertNotNull(cache.getChildren(tableZPath));
+      assertTrue(cache.childrenCached(tableZPath));
+
+      client.tableOperations().delete(tableName);
+      Wait.waitFor(() -> tableDeletedEvent.get(), 60_000);
+      assertFalse(cache.childrenCached(tableZPath)); // ZooCache watcher fired and deleted the data
+      assertNull(cache.getChildren(tableZPath));
+
+      assertFalse(connectionOpenEvent.get());
+      assertFalse(connectionClosedEvent.get());
+      tableDeletedEvent.set(false);
+
+      getCluster().getClusterControl().stop(ServerType.ZOOKEEPER);
```

Comment:
I like this test.  In a follow on we could explore pausing the ZK process using SIGSTOP and SIGCONT in addition to killing it.  Like pause the ZK process for 1.5x the configured ZK timeout and see if things behave as expected.

Not sure if this would change anything for this test, wondering if configuring the zookeeper timeout to lower than the default of 30s would help w/ runtimes for this test.